sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableDestinationCoder.java

```diff
@@ -45,13 +46,20 @@ public void encode(TableDestination value, OutputStream outStream)
     }
     tableSpecCoder.encode(value.getTableSpec(), outStream);
     tableDescriptionCoder.encode(value.getTableDescription(), outStream);
+    timePartitioningCoder.encode(value.getJsonTimePartitioning(), outStream);
   }
 
   @Override
   public TableDestination decode(InputStream inStream) throws IOException {
     String tableSpec = tableSpecCoder.decode(inStream);
     String tableDescription = tableDescriptionCoder.decode(inStream);
-    return new TableDestination(tableSpec, tableDescription);
+    String jsonTimePartitioning = null;
+    try {
+      jsonTimePartitioning = timePartitioningCoder.decode(inStream);
+    } catch (IOException e) {
+      // This implies we're decoding old state that did not contain TimePartitioning. Continue.
```

Comment:
Making this incompatible I think is a non starter. This would be very bad for the Dataflow runner, and this sink is today primarily used by the the Dataflow runner.

I'm not sure how adding an extra boolean helps. That boolean won't be serialized by previous versions of the coder, so won't we need the exact same code to attempt to deserialize the boolean and fall back to false if that fails?